### DISCUSSION TITLE
Action on clicked table row (based on SQL view)

### DISCUSSION THREAD

**Discussion by hjbout** [Budibase/budibase#5231] [Help] [answered]
Hi, I have 2 MySQL tables and a view based upon those 2 tables. The view is displayed as a table(block). I want the user to select one row (and only one). Upon clicking the row I want to add a row to one of the tables (not the view) with some data of the clicked row. I can link to another screen but I want to do some actions (save row). 
I can’t figure out how to accomplish this. Any suggestions?

**Comment by melohagan** [comment, marked as answer]
Hi @hjbout 

At the moment you cannot define actions for **Link table rows**. This only allows you to be directed to a screen.

However, what you can do is replace your table block with a Data Provider > Table combo:
![Screenshot 2022-04-04 at 12 31 23](https://user-images.githubusercontent.com/101575380/161534955-74f19fdc-a338-4a30-aa84-d5824112e88b.png)

Then you can click on the table, and select Elements > Button
![Screenshot 2022-04-04 at 12 33 10](https://user-images.githubusercontent.com/101575380/161535220-1b49e427-98fc-4243-8719-9e07078f97f3.png)

This will add a button for each row. You can then **Define actions** > **Save Row**.

Data source: none
Table: destination

Then click **Add Column** for all the column data you want to save. Here you can specify bindings that will be able to grab data from the current row.

**Reply by hjbout** [reply]
Thanks @melohagan, that helped! I'm getting there... slowly :)
